api/src/main/java/org/vivoweb/webapp/controller/freemarker/CreateAndLinkResourceController.java

```diff
@@ -0,0 +1,1910 @@
+                    vcard.addProperty(model.createProperty(VCARD_HAS_NAME), name);
+                    name.addProperty(RDF.type, model.getResource(VCARD_NAME));
+                    if (!StringUtils.isEmpty(author.given)) {
+                        name.addProperty(model.createProperty(VCARD_GIVENNAME), author.given);
+                    }
+                    if (!StringUtils.isEmpty(author.family)) {
+                        name.addProperty(model.createProperty(VCARD_FAMILYNAME), author.family);
+                    }
+
+                    Resource authorship = model.createResource(getUnusedUri(vreq));
+                    authorship.addProperty(RDF.type, model.getResource(VIVO_AUTHORSHIP));
+
+                    authorship.addProperty(model.createProperty(VIVO_RELATES), model.getResource(vivoUri));
+                    authorship.addProperty(model.createProperty(VIVO_RELATES), model.getResource(vcard.getURI()));
+
+                    model.getResource(vivoUri).addProperty(model.createProperty(VIVO_RELATEDBY), authorship);
+                    vcard.addProperty(model.createProperty(VIVO_RELATEDBY), authorship);
+                    authorship.addLiteral(model.createProperty(VIVO_RANK), rank);
+                }
+                rank++;
+            }
+        }
+
+        // Add a URL
+        if (!StringUtils.isEmpty(resourceModel.URL)) {
+            try {
+                URL url = new URL(resourceModel.URL);
+                Resource urlModel = model.createResource(getUnusedUri(vreq));
+                urlModel.addProperty(RDF.type, model.getResource(VCARD_URL_CLASS));
+                urlModel.addLiteral(model.createProperty(VCARD_URL_PROPERTY), url);
+
+                Resource kindModel = model.createResource(getUnusedUri(vreq));
+                kindModel.addProperty(RDF.type, model.getResource(VCARD_KIND));
+                kindModel.addProperty(model.createProperty(VCARD_HAS_URL), urlModel);
+                kindModel.addProperty(model.createProperty(OBO_CONTACT_INFO_FOR), work);
+
+                work.addProperty(model.createProperty(OBO_HAS_CONTACT_INFO), kindModel);
+            } catch (MalformedURLException e) {
+            }
+        }
+
+        // editor
+        // translator
+        // subject
+        // status
+        // presented at
+        // keyword
+
+        // http://purl.org/ontology/bibo/status
+        // http://vivoweb.org/ontology/core#hasSubjectArea
+        // http://purl.org/ontology/bibo/presentedAt
+        // http://vivoweb.org/ontology/core#freetextKeyword
+        // http://purl.org/ontology/bibo/translator (Direct to user)
+
+        // http://vivoweb.org/ontology/core#Editorship
+
+        return vivoUri;
+    }
+
+    /**
+     * Get a URI for the publisher object
+     *
+     * @param vreq
+     * @param publisher
+     * @return
+     */
+    protected String getPublisherURI(VitroRequest vreq, String publisher) {
+        return getUnusedUri(vreq);
```

Comment:
This (and getVCardURI()) had bigger implementations in OpenVIVO because it was opinionated about how to generate URIs in different contexts (ORCID IDs, DOIs, etc.) - however, this also required other modifications in the code to handle the URIs.

In the current state of VIVO - without those modifications, when all the URIs generated fit the nXXXX pattern - it doesn't make sense to have anything other than using a URI that fits the pattern right now.